### PR TITLE
Scan suggest message when compose build / compose up

### PR DESCRIPTION
**Pull request by gtardif**
**What I did**
* Add scan suggest message at the end of `docker compose build`, and end of build section when `docker compose up`
* check scan plugin is available before displaying message
* Allow users to turn this off setting `DOCKER_SCAN_SUGGEST = false`
* Do not display message if user has already used scan command

```
$ ./bin/docker compose -f ./local/e2e/compose/fixtures/build-test/compose.yml build
...
=> [build-test_nginx] exporting to image  
=> => exporting layers            
=> => writing image sha256:f727547da0e5aa30472c3705fd70bfe02ce300e50be0665eca05b6482bdd1aa9        
=> => naming to docker.io/gtardif/custom-nginx:1.0 
=> => naming to docker.io/library/build-test_nginx       
Try scanning the image you have just built to identify vulnerabilities with Docker’s new security tool: 
docker scan myorg/custom-nginx:1.0
docker scan build-test_nginx
$
```

and compose up: 
```
$ ./bin/docker compose -f ./local/e2e/compose/fixtures/build-test/compose.yml up -d
...
=> => writing image sha256:f727547da0e5aa30472c3705fd70bfe02ce300e50be0665eca05b6482bdd1aa9
=> => naming to docker.io/library/build-test_nginx                 
=> => naming to docker.io/library/custom-nginx    
Try scanning the image you have just built to identify vulnerabilities with Docker’s new security tool: 
docker scan build-test_nginx
docker scan custom-nginx
[+] Running 3/3
 ⠿ Network "build-test_default"   Created    
 ⠿ Container build-test_nginx_1   Started    
 ⠿ Container build-test_nginx2_1  Started   
$    
```

**Related issue**

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: local/compose/scan_suggest.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		return
+	}
+	if !scanAvailable() || scanAlreadyInvoked(ctx) {
+		return
+	}
+	commands := []string{}
+	for _, image := range builtImages {
+		commands = append(commands, fmt.Sprintf("docker scan %s", image))
+	}
+	allCommands := strings.Join(commands, ", ")
```
Comment (aiordache) [inline local/compose/scan_suggest.go:47]: Maybe output one scan command per line. Having all commands on one line would be confusing (unless we can pass all images to one scan command).

File: local/compose/scan_suggest.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	if os.IsNotExist(err) {
+		return false
+	}
+	if f.IsDir() { // should never happen, do not bother user with suggestion if something goes wrong
+		return true
+	}
+	type scanOptin struct {
+		Optin bool `json:"optin"`
+	}
+	data, err := ioutil.ReadFile("/tmp/dat")
```
Comment (aiordache) [inline local/compose/scan_suggest.go:63]: Should this filepath be `scan/config.json`?

Comment (gtardif) [reply local/compose/scan_suggest.go:63]: outch, of course, cp/paste error...

File: local/compose/scan_suggest.go (hunk 1 above)
Comment (gtardif) [reply local/compose/scan_suggest.go]: updated with this

**Review by aiordache — APPROVED** [review]

**Comment by gtardif** [comment]
Updated message after sync/confirmation by various teams: 'Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them'

**Pull request merged**